### DISCUSSION TITLE
Permissions on Page level

### DISCUSSION THREAD

**Discussion by LMSSonos** [OrchardCMS/OrchardCore#13487] [Q&A] [answered]
Hello

We are investigating to use OrchardCore for an intranet. The idea is that all users have to authenticate to view the webpage. The question is, if permissions can be set, that some pages from the intranet are only visible to a specific user-group. 

I did not see anything specific but asking myself if there is a liquid-tag to verify the users permissions and redirect in case he has no access.

Also for the menu this would be used to display the entry or not based on the users permissions

Thanks

**Comment by hishamco** [comment, marked as answer]
You can use `IAuthorizationService.AuthorizeAsync()` to check a certain permission in the code or UI

**Comment by ns8482e** [comment]
You can inject `IAuthorizeService` in your views and based on authorization result from `AuthorizeAsync` you can show/hide in views.

However IMHO redirecting from views will be too late. To redirect based on permission - either use controller or IActionFilter

**Comment by LMSSonos** [comment]
@hishamco and @ns8482e  - thanks a lot for your shared input. I will make a PoC in the next weeks and come back with the results!

**Reply by hishamco** [reply]
@LMSSonos I don't how my answer is different from Niraj :)

**Reply by hishamco** [reply]
@LMSSonos I don't how my answer is different from Niraj :)

**Reply by LMSSonos** [reply]
sorry my bad, but I wanted to mark both as answers but that's not possible, so I change it ;-)
